gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -198,6 +200,10 @@ public void launchJob(JobListener jobListener) throws JobException {
             "Previous instance of job %s is still running, skipping this scheduled run", this.jobContext.getJobName()));
       }
 
+      if (this.jobContext.getSemanticsType() == DeliverySemantics.Type.EXACTLY_ONCE) {
+        executeUnfinishedCommitSequences(jobState.getJobName());
```

Comment:
To meet the requirement of exactly-once, if the previous run's commit sequence failed to complete, the next run should not make further progress.